### PR TITLE
Documentation renewal

### PR DESCRIPTION
**Pull request by Guts**
# Description

As already mentioned [here](https://github.com/opengisch/qgis-plugin-ci/issues/31#issuecomment-748495292) and [there](https://github.com/opengisch/qgis-plugin-ci/pull/33#issuecomment-775869328), I think that the README file alone is no longer sufficient in view of all the possible options and usage scenarios.

# Changes

- move README content into multiples sections below the well-named `docs` folder: configure, use, contribute.
- use MyST parser to write in Markdown with some additional sugar-features
- use Sphinx to build the output (mainly HTML)
- stores the specific requirements into `requirements/documentation.txt` to improve readability and maintenability of the project
- add an about module to expose project metadata through code and documentation
- add GitHub Actions to build and deploy documentation to gh-pages branch
- add dependabot configuration to automate dependencies watching

Here is a preview of the final rendering (not contractual):

Screenshot_2021-02-10 QGIS Plugin CI - Documentation — QGIS Plugin CI 1 8 4 documentation

# Decisions

If you agree to this, I need you give your opinion on several points:

- [x] where to publish documentation: GitHub Pages :heavy_check_mark:  or ~~Read The Docs~~? (or another option but I'm mostly comfortable with those two)
- [x] I've added the name of @Gustry  and mine as contributors. Are you agree with that @3nids? - :heavy_check_mark: 
- [x] do you have a logo for the project or any image to use as? No :white_check_mark: 

# Notes

:information_source:  Work done during Open Source time offered as [Oslandia](https://oslandia.com/) teammate. :heart:

### CODE REVIEW

**Comment by 3nids** [comment]
Wow, awesome work!
I'd vote for GitHub pages, but no strong opinion. Whatever sounds best.
Of course, your name is more than welcome in the list.
I don't have any logo at the moment.

File: docs/usage/ci_travis.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+```yaml
+- provider: releases
+  name: Title of the release ${TRAVIS_TAG}
+  api_key: ${GH_TOKEN}
+  on:
+    tags: true
+```
+
+## Submodules
```
Comment (Gustry) [inline docs/usage/ci_travis.md:35]: This is not specific to travis.

File: docs/usage/ci_travis.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+before_install:
+  # cannot use SSH to fetch submodule
+  - sed -i 's#git@github.com:#https://github.com/#' .gitmodules
+  - git submodule update --init --recursive
+```
+
+When packaging the plugin, it's possible to not update the submodule using CLI options.
+
+## Using Transifex to translate your plugin
```
Comment (Gustry) [inline docs/usage/ci_travis.md:51]: Same, this is not specific to travis.

**Review by Gustry — APPROVED** [review]
LGTM thanks !
TIL (for me) ;-)

**Comment by Guts** [comment]
> Wow, awesome work!

Thanks :). Done on my "open source time" at Oslandia ;).

> I'd vote for GitHub pages, but no strong opinion. Whatever sounds best.

> ok. I've pushed the workflow to deploy on gh-pages. I let you configure this on repo settings.

> Of course, your name is more than welcome in the list.

Thanks.

I've updated the PR description.

File: docs/development/history.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+```{include} ../../CHANGELOG.md
```
Comment (Gustry) [inline docs/development/history.md:1]: This changelog is for testing purpose, not sure it's worth it to include it here.
Or did you plan to tweak the test and have a proper one for the project ?

File: docs/development/documentation.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
+# Documentation
+
+Project uses Sphinx to generate documentation from docstrings (documentation in-code) and custom pages written in Markdown (througth the [MyST parser](https://myst-parser.readthedocs.io/en/latest/)).
+
+## Build documentation website
+
+To build it:
+
+```bash
+# install aditionnal dependencies
```
Comment (Gustry) [inline docs/development/documentation.md:10]:
```suggestion
# install additional dependencies
```

File: docs/development/documentation.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# Documentation
+
+Project uses Sphinx to generate documentation from docstrings (documentation in-code) and custom pages written in Markdown (througth the [MyST parser](https://myst-parser.readthedocs.io/en/latest/)).
```
Comment (Gustry) [inline docs/development/documentation.md:3]:
```suggestion
Project uses Sphinx to generate documentation from docstrings (documentation in-code) and custom pages written in Markdown (through the [MyST parser](https://myst-parser.readthedocs.io/en/latest/)).
```

**Review by Gustry — COMMENTED** [review]
I tried on my local quickly, nice ;-)

File: docs/development/history.md (hunk 3 above)
Comment (Guts) [reply docs/development/history.md:1]: You've read my mind ;)

File: docs/usage/ci_travis.md (hunk 1 above)
Comment (Guts) [reply docs/usage/ci_travis.md]: Removed.

File: docs/usage/ci_travis.md (hunk 2 above)
Comment (Guts) [reply docs/usage/ci_travis.md]: Moved into translation.md page

**Comment by Guts** [comment]
@Gustry @3nids I'm done here but I don't understand why tests are failing.

**Comment by 3nids** [comment]
The tests are failing since we are running from a fork and the secrets are not shared.
We could use the `pull_request_target`, does it sound safe to you in our scenario? https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request_target

**Comment by Gustry** [comment]
Thanks for the merge.

Side note @Guts It's now native in github https://twitter.com/GHchangelog/status/1358961977629024258

**Comment by 3nids** [comment]
> We could use the pull_request_target, does it sound safe to you in our scenario? 
no not safe, as we run `setup.py` in the tests.

**Comment by Guts** [comment]
@3nids now the documentation is fixed and deployed to gh-pages branch, could you enable it into repository settings please?

**Comment by 3nids** [comment]
Done!

**Comment by Guts** [comment]
> Done!

Yeah! Here we go: https://opengisch.github.io/qgis-plugin-ci/

**Pull request merged**